### PR TITLE
Added platform registration and data-exchange logic.

### PR DESCRIPTION
**Pull request by Shico-UA**
Working registration and data exchange logic.

### CODE REVIEW

**Comment by auraz** [comment]
Please find better name :) You can ask for ideas in some channel :) inside our slack

File: .gitignore [hunk 1]
```diff
@@ -93,3 +93,6 @@ bin/
 include/
 pip-selfcheck.json
 requirement.txt
+
+migrations/
+*.sqlite3
```
Comment (auraz) [inline .gitignore]: missed newline

File: graph_creator/views.py [hunk 2]
```diff
@@ -3,10 +3,10 @@
...
 from django.core import serializers
 from django.shortcuts import render
+from django.shortcuts import redirect
 from django.views.decorators.csrf import csrf_exempt
-from django.utils.decorators import method_decorator
 from django.views.generic import View
-from django.shortcuts import redirect
+from django.utils.decorators import method_decorator
 from django.urls import reverse
 from .models import DataStorage
```
Comment (auraz) [inline graph_creator/views.py]: missed new line (imports according to PEP8)

File: graph_creator/views.py [hunk 3]
```diff
@@ -19,9 +19,15 @@ def get(self, request, *args, **kwargs):
 
 @method_decorator(csrf_exempt, name='dispatch')
 class ReceiveData(View):
+    """
+    This view receives data from the remote edx-platform.
```
Comment (auraz) [inline graph_creator/views.py]: Please follow open edx docstring style

File: graph_creator/views.py [hunk 4]
```diff
@@ -19,9 +19,15 @@ def get(self, request, *args, **kwargs):
...
+    If the platform has already registered, a normal data exchange happens.
+    Otherwise we generate a secret token, save it to DB with the edx-platform incoming URL
+    and send newly generated token to the edx-platform for further registration
+    and data interchange abilities with the server.
+    """
     def post(self, request, *args, **kwargs):
```
Comment (auraz) [inline graph_creator/views.py]: Add function docstring

File: graph_creator/views.py [hunk 5]
```diff
@@ -30,22 +36,21 @@ def post(self, request, *args, **kwargs):
         platform_url = received_data.get('platform_url')
         secret_token = received_data.get('secret_token')
 
-        # Token check and data saving
-        if secret_token and isinstance(secret_token, uuid.UUID):
-            obj, created = DataStorage.objects.get_or_create(secret_token=secret_token)
+        if secret_token is None:
+            secret_token = uuid.uuid4().hex
+            obj = DataStorage.objects.create()
```
Comment (auraz) [inline graph_creator/views.py]: you can create with kwargs. Use this form please.

File: graph_creator/views.py [hunk 6]
```diff
@@ -30,22 +36,21 @@ def post(self, request, *args, **kwargs):
...
+            obj.secret_token = secret_token
+            obj.platform_url = platform_url
+            obj.save()
+            reverse_token = requests.post('http://192.168.1.139:8000/acceptor_data/',
+                                          data={"reverse_token": secret_token})
```
Comment (auraz) [inline graph_creator/views.py]: Please follow open edx docstring style

File: .gitignore (hunk 1 above)
Comment (Shico-UA) [inline .gitignore]: Done

File: graph_creator/views.py (hunk 2 above)
Comment (Shico-UA) [inline graph_creator/views.py]: Done

File: graph_creator/views.py (hunk 3 above)
Comment (Shico-UA) [inline graph_creator/views.py]: Done

File: graph_creator/views.py (hunk 4 above)
Comment (Shico-UA) [inline graph_creator/views.py]: Done

File: graph_creator/views.py (hunk 5 above)
Comment (Shico-UA) [inline graph_creator/views.py]: Done

File: graph_creator/views.py (hunk 6 above)
Comment (Shico-UA) [inline graph_creator/views.py]: Done

**Pull request closed without merging**